### PR TITLE
[DataTable]: columns group.

### PR DESCRIPTION
**Pull request by Kuznietsov**
### Summary

Added column groups to DataTable.

### CODE REVIEW

File: uui/components/tables/DataTableHeaderCell.module.scss [hunk 1]
```diff
@@ -20,7 +20,7 @@
     padding-inline-end: var(--uui-dt-header-cell-padding-end);
     width: 0;
     background-clip: padding-box;
-    height: var(--uui-dt-header-cell-height);
+    min-height: var(--uui-dt-header-cell-height);
```
Comment (Kuznietsov) [inline uui/components/tables/DataTableHeaderCell.module.scss:23]: It is required for DataTableHeaderCell to expand to 100% height if groups are enabled.

File: uui-core/src/hooks/useColumnGroups.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        columns.reduce<Record<string, DataColumnGroupProps>>((vGroups, column) => {
+            if (column.group) {
+                vGroups[column.group] = groupsRecord[column.group];
+                return vGroups;
+            }
+            return vGroups;
+        }, {}),
+    ), [groupsRecord, columns]);
+
+    return { groups: visiblGroups };
```
Comment (jakobz) [inline uui-core/src/hooks/useColumnGroups.ts:21]: visibl**e**Groups

File: uui-core/src/hooks/useColumnGroups.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { useMemo } from 'react';
+import { DataColumnGroupProps, DataColumnProps } from '../types';
+
+export function useColumnGroups<TItem, TId>(groups: DataColumnGroupProps[], columns: DataColumnProps<TItem, TId>[]) {
```
Comment (jakobz) [inline uui-core/src/hooks/useColumnGroups.ts:4]: check if needed

File: uui/components/tables/DataTable.tsx [hunk 4]
```diff
@@ -26,6 +28,8 @@ export interface DataTableProps<TItem, TId, TFilter = any> extends IEditable<Dat
     /** Rows that should be rendered in table */
     rows?: DataRowProps<TItem, TId>[];
 
+    groups?: DataColumnGroupProps[];
```
Comment (jakobz) [inline uui/components/tables/DataTable.tsx:31]: => columnGroups

**Pull request merged**